Create rule S7155: CircleCI secrets should not be disclosed (APPSEC-2239) (#4490)
SonarSource/rspec: rules/S7155/secrets/rule.adoc

include::../../../shared_content/secrets/description.adoc[]

> If attackers gains access to a CircleCI API key, they might be able to modify projects and jobs running on the CircleCI platform.

Reviewer: ```suggestion
If attackers gain access to a CircleCI API token, they might be able to modify projects and jobs running on the CircleCI platform.
```

Related to previous change. Also, small typo.

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

> The exact impact of the compromise of an CircleCI API key varies depending on the permissions granted to this token and of its type (personal or project token). It can range from loss of sensitive data and source code to severe supply chain attacks.

Reviewer: ```suggestion
The exact impact of the compromise of a CircleCI API token varies depending on the permissions granted to this token and of its type (personal or project token). It can range from loss of sensitive data and source code to severe supply chain attacks.
```

Below are some real-world scenarios that illustrate some impacts of an attacker
exploiting the secret.

include::../../../shared_content/secrets/impact/source_code_compromise.adoc[]

include::../../../shared_content/secrets/impact/supply_chain_attack.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

:example_secret: CCIPAT_FERZRjTN451xnDCy1y9gWn_79fb6ca4d0e5f833612eee17de397a9dca0a9e9f

> :example_name: cci-api-key
> :example_env: CCI_API_KEY

Reviewer: ```suggestion
:example_name: cci-api-token
:example_env: CCI_API_TOKEN
```
